src/java/com/sarality/app/view/BaseViewInitializer.java

```diff
@@ -64,19 +69,50 @@ public void init(DataSource<T> dataSource) {
     activity.getSupportLoaderManager().initLoader(0, null, new DataLoader<T>(activity, dataSource, this));
   }
 
+  protected boolean assertValidAction(ViewAction action) {
+    // Check if the view exists
+    int viewId = action.getViewId();
+    View actionView = view.findViewById(viewId);
+    if (actionView == null) {
+      String viewIdName = activity.getResources().getResourceEntryName(viewId);
+      throw new IllegalArgumentException("View with id " + viewIdName + " not found while registering action "
+          + action.getClass().getSimpleName());
+    }
+
+    // Check if the trigger type supported on the view
+    TriggerType triggerType = action.getTriggerType();
+    if (!getSupportedTriggerTypes().contains(triggerType)) {
+      String viewIdName = activity.getResources().getResourceName(viewId);
+      throw new IllegalArgumentException("Cannot register a " + triggerType + " actions for the view with id "
+          + viewIdName + ". Only the following triggers are supported on the view " + getSupportedTriggerTypes());
+    }
+
+    // Check if an action with the trigger type already exists on the view
```

Comment:
why should this not be allowed? This was the purpose of the composite view that we should be able to do multiple actions with the same trigger type on the same view.